File: day-01/lua/part2.lua
#!/usr/bin/env lua5.4

local function parseInput()
    local num = {}
    for str in io.lines(arg[1]) do
        num[#num+1] = tonumber(str)
    end
    return num
end

local num = parseInput()
local target = 2020
-- local target = 99920044

local function part2()
    for i=1, #num do
        for j=1, #num do
            for k=1, #num do
                if num[i] + num[j] + num[k] == target then
                    return num[i] * num[j] * num[k]
                end
            end
        end
    end
end

print(print(string.format("%.0f",part2())))

File: day-01/python/part2.py
#!/usr/bin/env python3

import sys

num = [int(i) for i in open(sys.argv[1], 'r').read().split("\n")]
target = 2020
# target = 99920044

def part2():
    for i in num:
        for j in num:
            for k in num:
                if i+j+k == target:
                    assert i*j*k == 96047280
                    return [i, j, k, i*j*k]

result = part2()
print(f"{result[0]} * {result[1]} * {result[2]} = {result[3]}")
